fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintenanceTest.java

```diff
@@ -781,6 +784,85 @@ public void ensureValid() throws IOException {
         }
     }
 
+    @Test
+    void concurrentUpdate() throws IOException {
+        // Once the two issues noted below are fixed, we should make this parameterized, and run with additional random
+        // configurations.
+        AtomicInteger threadCounter = new AtomicInteger();
+        // Synchronization blocks in FDBDirectoryWrapper can cause a deadlock
+        // see https://github.com/FoundationDB/fdb-record-layer/issues/2989
+        // So set the pool large enough to overcome that
+        this.dbExtension.getDatabaseFactory().setExecutor(new ForkJoinPool(30,
```

Comment:
Does that have to be cleaned at the end of the test or is the dbFactory getting a new executor on every test?

Reply:
It does not. The https://github.com/FoundationDB/fdb-record-layer/blob/5f21535c68bf397bf7636335b00cfccbc72878fa/fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/test/FDBDatabaseExtension.java#L154 clears out the database factory after each test, so that each test has it's own factory, and thus they can test things like this without affecting others, either serially or in parallel.